server/events/vcs/gitlab_client.go

```diff
@@ -355,18 +355,28 @@ func (g *GitlabClient) PullIsMergeable(logger logging.SimpleLogging, repo models
 		return false, err
 	}
 
+	if supportsDetailedMergeStatus {
+		logger.Debug("Detailed merge status: '%s'", mr.DetailedMergeStatus)
+	} else {
+		logger.Debug("Merge status: '%s'", mr.MergeStatus) //nolint:staticcheck // Need to reference deprecated field for backwards compatibility
+	}
+
 	if ((supportsDetailedMergeStatus &&
 		(mr.DetailedMergeStatus == "mergeable" ||
 			mr.DetailedMergeStatus == "ci_still_running" ||
-			mr.DetailedMergeStatus == "ci_must_pass")) ||
+			mr.DetailedMergeStatus == "ci_must_pass" ||
+			mr.DetailedMergeStatus == "need_rebase")) ||
 		(!supportsDetailedMergeStatus &&
 			mr.MergeStatus == "can_be_merged")) && //nolint:staticcheck // Need to reference deprecated field for backwards compatibility
 		mr.ApprovalsBeforeMerge <= 0 &&
 		mr.BlockingDiscussionsResolved &&
 		!mr.WorkInProgress &&
 		(allowSkippedPipeline || !isPipelineSkipped) {
```

Comment:
This condition might be easier to maintain in a function?

Reply:
Also `ci_still_running`, `ci_must_pass`, `need_rebase` and `can_be_merged` can be converted to constants.

Reply:
Thanks for the review @GMartinez-Sisti. I agree with your comments, but as this is just a simple fix, I am keeping the code in its current pattern. If we want to refactor this function, it can be done in a separate PR.